gwpy/signal/spectral/_scipy.py

```diff
@@ -200,6 +206,34 @@ def coherence(timeseries, other, segmentlength, noverlap=None, **kwargs):
     --------
     scipy.signal.coherence
     """
+
+    # Should we warn about unequal sampling frequencies?
+    warn_fs = False
+
+    if downsample is None:
+        warn_fs = True
+        downsample = True
+
+    if timeseries.sample_rate != other.sample_rate:
+        # scipy assumes a single sampling frequency
+        if not downsample:
+            raise ValueError("Cannot calculate coherence when sampling "
+                             "frequencies are unequal")
+        else:
```

Comment:
Since the `if not downsample` clause raises an exception, the `else` is not required, so that can be removed and the whole of its contents just dedented by one level.

Reply:
Removed as requested